### PR TITLE
adding reopen_duration feature

### PR DESCRIPTION
**Pull request by raikakhodadad1**
A new feature that was necessary for my use case while trying to keep the original functionality in place. 

Added functionality so the user can set the number of days to determine whether a ticket with same labels should be re-opened or recreated. If the new value introduced is set to 0, it will keep the original functionality of re-opening the same ticket. 

I have tested this over couple of months and it is working as expected in my use case.

### CODE REVIEW

**Comment by free** [comment]
Oh, and one more issue: [the Travis CI build is failing](https://travis-ci.org/free/jiralert/builds/451629593?utm_source=github_status&utm_medium=notification) because of a deprecated (due to your updating the go-jira dependency) way of authenticating with JIRA. The fix appears to be straightforward, [according to the go-jira readme](https://github.com/andygrunwald/go-jira#basic-auth-example).

And thanks for contributing. I should have started with that.

**Comment by raikakhodadad1** [comment]
Included the recommendations suggested. Also made couple of minor changes in other places: 
1) line 95 in notify.go - I needed the component to be read from the template so I modified the code to allow that
2) line 219/223 on notify.go I made a new pointer and repointed because the value of the pointer was getting lost previously and the ticket number would never print out

File: notify.go [hunk 1]
```diff
@@ -60,8 +63,12 @@ func (r *Receiver) Notify(data *alertmanager.Data) (bool, error) {
 			log.Infof("Issue %s for %s is resolved as %q, not reopening", issue.Key, issueLabel, issue.Fields.Resolution.Name)
 			return false, nil
 		}
-		log.Infof("Issue %s for %s was resolved, reopening", issue.Key, issueLabel)
-		return r.reopen(issue.Key)
+
+		resolutionTime := time.Time(issue.Fields.Resolutiondate)
+		if resolutionTime.Add(time.Duration(*r.conf.ReopenDuration)).After(time.Now()) {
+			log.Infof("Issue %s for %s was resolved on %s, reopening", issue.Key, issueLabel, resolutionTime.Format(time.UnixDate))
```
[the remark this answers is not in the document]
Comment (raikakhodadad1) [reply notify.go]: It's no problem I'll change this and resubmit - totally fair about it being confusing.

File: notify.go [hunk 2]
```diff
@@ -208,10 +216,11 @@ func (r *Receiver) reopen(issueKey string) (bool, error) {
 
 func (r *Receiver) create(issue *jira.Issue) (bool, error) {
 	log.V(1).Infof("create: issue=%v", *issue)
-	issue, resp, err := r.client.Issue.Create(issue)
+	newIssue, resp, err := r.client.Issue.Create(issue)
```
[the remark this answers is not in the document]
Comment (raikakhodadad1) [reply notify.go]: Actually yep! Sorry I didn't explain in details. So in creating new issues, the ticket number would never print to stdout when I was testing. After tracing it I found out that it was a pointer issue. The old code was trying to change the memory address that the pointer pointed to which wasn't working. So to make sure the ticket number always gets printed out I just created a new value to store the data and then repointed.

File: notify.go [hunk 3]
```diff
@@ -166,8 +173,8 @@ func toIssueLabel(groupLabels alertmanager.KV) string {
 func (r *Receiver) search(project, issueLabel string) (*jira.Issue, bool, error) {
 	query := fmt.Sprintf("project=\"%s\" and labels=%q order by key", project, issueLabel)
```
[the remark this answers is not in the document]
Comment (raikakhodadad1) [reply notify.go]: I originally had this sorted by resolutiondate per your suggestion however in a use case where two tickets would be open simultaneously and they would get resolved not in order this wouldn't work (example ticket.123 opens first, ticket.234 opens second after reopen_duration has passed but ticket.234 gets closed before ticket.123). That's why I ordered based on the key. Although it should still be ordered by desc. In my original testing it seemed that jira always returned the higher key value but testing again today that seems to not be true. I will resubmit with "order by key desc". Let me know if you still think I should order by resolution date?

**Comment by raikakhodadad1** [comment]
Sorry it took me a while to respond but changed and tested locally with ordering by resolutiondate.

**Comment by free** [comment]
There are a couple of static check failures, but they're all mine, so I get to fix them. Thanks for contributing and putting up with my review comments.

**Pull request merged**